python/Converters/PycBasicData.h

```diff
@@ -99,19 +99,24 @@ namespace casacore { namespace python {
       PyObject* obj_ptr,
       boost::python::converter::rvalue_from_python_stage1_data* data)
     {
-#ifdef IS_PY3K
-    PyObject * temp_bytes = PyUnicode_AsEncodedString(obj_ptr, "ASCII", "strict"); // Owned reference
-    char* value;
-    if (temp_bytes != NULL) {
-        value = PyBytes_AS_STRING(temp_bytes); // Borrowed pointer
-        value = strdup(value);
-        Py_DECREF(temp_bytes);
+    char* value = 0;
+    if (PyUnicode_Check(obj_ptr)) {
+        PyObject * temp_bytes = PyUnicode_AsEncodedString(obj_ptr, "UTF-8", "strict"); // Owned reference
+
+        if (temp_bytes != NULL) {
+            value = PyBytes_AS_STRING(temp_bytes); // Borrowed pointer
+            value = strdup(value);
```

Comment:
I think this copy is going to leak (unless the String constructor takes ownership, but in that case it's going to steal ownership from PyString_AsString in Python 2).

Reply:
well spotted. I'm not sure what to do here, it is a bug in the original code. What do you propose?

Reply:
I would do the following
- create a Boost.Python wrapper around `temp_bytes` to ensure that it gets decref'ed once it goes out of scope (you'd need to check the syntax to ensure that the wrapper borrows the reference rather than adding its own).
- Put that wrapper at a sufficient outer scope so that it lives past creation of the `String`.
- Get rid of the `strdup`.

That should not only fix the leak, but eliminate a copy so that the `String` is constructed directly from the data in `temp_bytes`.

Reply:
opened a new issue for this. Thanks Bruce! https://github.com/casacore/casacore/issues/908